java/src/gov/usgs/processingformats/TravelTimePlotDataBranch.java

```diff
@@ -0,0 +1,263 @@
+package gov.usgs.processingformats;
+
+import java.util.*;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+
+import gov.usgs.processingformats.ProcessingInt;
+
+/**
+ * a conversion class used to create, parse, and validate travel time plot
+ * branch data
+ * 
+ * @author U.S. Geological Survey &lt;jpatton at usgs.gov&gt;
+ */
+public class TravelTimePlotDataBranch implements ProcessingInt {
+
+	/**
+	 * JSON Keys
+	 */
+	public static final String PHASE_KEY = "Phase";
+	public static final String SAMPLES_KEY = "Samples";
+
+	/**
+	 * Required seismic phase code for this branch
+	 */
+	private String phase;
+
+	/**
+	 * A required ArrayList of TravelTimePlotDataSample objects
+	 */
+	private ArrayList<TravelTimePlotDataSample> samples;
+
+	/**
+	 * The constructor for the TravelTimePlotDataBranch class. Initializes
+	 * members to null values.
+	 */
+	public TravelTimePlotDataBranch() {
+
+		reload(null, null);
+	}
+
+	/**
+	 * Advanced constructor
+	 * 
+	 * The advanced constructor for the TravelTimeData class. Initializes
+	 * members to provided values.
+	 * 
+	 * @param newPhase
+	 *            - A String containing the seismic phase code
+	 * @param newSamples
+	 *            - A ArrayList&lt;TravelTimePlotDataSample&gt; containing the
+	 *            sample data
+	 */
+	public TravelTimePlotDataBranch(String newPhase,
+			ArrayList<TravelTimePlotDataSample> newSamples) {
+
+		reload(newPhase, newSamples);
+	}
+
+	/**
+	 * Constructs the class from a JSONObject, populating members
+	 * 
+	 * @param newJSONObject
+	 *            - A JSONObject.
+	 */
+	public TravelTimePlotDataBranch(JSONObject newJSONObject) {
+
+		// Required values
+		// phase
+		if (newJSONObject.containsKey(PHASE_KEY)) {
+			phase = newJSONObject.get(PHASE_KEY).toString();
+		} else {
+			phase = null;
+		}
+
+		// samples
+		if (newJSONObject.containsKey(SAMPLES_KEY)) {
+
+			samples = new ArrayList<TravelTimePlotDataSample>();
+
+			// get the array
+			JSONArray samplesArray = (JSONArray) newJSONObject.get(SAMPLES_KEY);
+
+			if ((samplesArray != null) && (!samplesArray.isEmpty())) {
+
+				// go through the whole array
+				for (int i = 0; i < samplesArray.size(); i++) {
+
+					// get the object
+					JSONObject samplesObject = (JSONObject) samplesArray.get(i);
+
+					// add to ArrayList
+					samples.add(new TravelTimePlotDataSample(samplesObject));
+
+				}
+			}
+		} else {
+			samples = null;
+		}
+	}
+
+	/**
+	 * Constructs the class from a TravelTimePlotData object, populating members
```

Comment:
more copy and paste this is the TravelTimePlotDataBranch class, same with next reload function